wpilibjExamples/src/main/java/edu/wpi/first/wpilibj/examples/romireference/commands/DriveDistance.java

```diff
@@ -0,0 +1,59 @@
+/*----------------------------------------------------------------------------*/
+/* Copyright (c) 2018-2020 FIRST. All Rights Reserved.                        */
+/* Open Source Software - may be modified and shared by FRC teams. The code   */
+/* must be accompanied by the FIRST BSD license file in the root directory of */
+/* the project.                                                               */
+/*----------------------------------------------------------------------------*/
+
+package frc.robot.commands;
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import frc.robot.subsystems.Drivetrain;
+
+public class DriveDistance extends CommandBase {
+  private final Drivetrain m_drive;
+  private final double m_distance;
+  private final double m_speed;
+  private double startingDistance;
+
+  /**
+   * Creates a new DriveDistance.
+   *
+   * @param speed The speed at which the robot will drive
+   * @param inches The number of inches the robot will drive
+   * @param drive The drivetrain subsystem on which this command will run
+   */
+  public DriveDistance(double speed, double inches, Drivetrain drive) {
+    m_distance = inches;
+    m_speed = speed;
+    m_drive = drive;
+    startingDistance = 0.0;
+    addRequirements(m_drive);
+  }
+
+   // Called when the command is initially scheduled.
+  @Override
+  public void initialize() {
+    m_drive.arcadeDrive(0, 0);
+    m_drive.resetEncoders();
+    //startingDistance = Math.abs(m_drive.getAverageDistanceInch());
+  }
+
+  // Called every time the scheduler runs while the command is scheduled.
+  @Override
+  public void execute() {
+    m_drive.arcadeDrive(m_speed, 0);
+  }
+
+  // Called once the command ends or is interrupted.
+  @Override
+  public void end(boolean interrupted) {
+    m_drive.arcadeDrive(0, 0);
+  }
+
+  // Returns true when the command should end.
+  @Override
+  public boolean isFinished() {
+    // Compare distance travelled from start to desired distance 
+    return (Math.abs(m_drive.getAverageDistanceInch()) - startingDistance) >= m_distance;
```

Comment:
We might want to make this a little more complex and also check that our rate is near zero.

Reply:
I thought I'd leave it basic here. I could add a rate check by moving this conditional to periodic, then setting speed to zero when the condition is met. Then in isFinished, I could check if completedDistance and prevDistance == currentDistance in the isFinished. I'll test it out and see if it's an easy patch.

Reply:
@AustinShalit , let me know if this is critical for this release or if you are good for me to add it later. For the most part, it's negligible since the motors are stopped right after it's finished. I was thinking of following this up with another PR, but at least make sure this gets in the release first.